pkg/plugins/skeleton/generic.go

```diff
@@ -218,9 +218,15 @@ func (p *PluginRegistrationWrapper) initialize() {
 	p.servers = make([]*grpc.Server, 0, len(p.sockets))
 }
 
-func (p *PluginRegistrationWrapper) start() error {
+func (p *PluginRegistrationWrapper) start() (startError error) {
 	p.Lock()
-	defer p.Unlock()
+	defer func() {
+		p.Unlock()
+		if startError != nil {
```

Comment:
when will the `startError` be set during the process?

Reply:
1. socket path provided in configuration is invalid
2. the wrapped plugin starts failed
3. any grpc server doesn't work

Reply:
i mean it seems `startError` has never been set in the below

Reply:
ok got it